jwst/datamodels/tests/test_models.py

```diff
@@ -49,6 +49,95 @@ def teardown():
     shutil.rmtree(TMP_DIR)
 
 
+@pytest.fixture(scope='module')
+def empty_model():
+    """Create an empy model"""
+    return DataModel()
+
+
+@pytest.fixture 
+def jail_environ():
+    """Lock cheanges to the environment"""
+    original = os.environ.copy()
+    try:
+        yield
+    finally:
+        os.environ = original
+
+
+@pytest.fixture(scope='module')
```

Comment:
This one probably also ought to be a function-scoped fixture.  We're only using it in one function anyway, so the module-scoping doesn't save us anything.

Reply:
This one is generating files, so it does not suffer the same issues as an open `DataModel` as before. And, though it is just being used in one test function, that function is parametrized and is re-executed, currently, 6 times.

As such, module scope should be appropriate.

Reply:
But since it's a module-scoped fixture, it only generates those files once _per module_, and re-uses the generated result for every test in the module.  Those 6 executions of the parametrized test are sharing the same files, is that the intention?

Reply:
Here's an example:

```python
import pytest

@pytest.fixture(scope='module')
def some_module_fixture():
    return {"values": []}

@pytest.mark.parametrize("value", [1, 2, 3])
def test_something(some_module_fixture, value):
    some_module_fixture["values"].append(value)
    assert some_module_fixture["values"] == [value]
```

The first of the parametrized tests succeeds, but the next two fail because the values start accumulating in the array:

```
some_module_fixture = {'values': [1, 2]}, value = 2

    @pytest.mark.parametrize("value", [1, 2, 3])
    def test_something(some_module_fixture, value):
        some_module_fixture["values"].append(value)
>       assert some_module_fixture["values"] == [value]
E       assert [1, 2] == [2]
E         At index 0 diff: 1 != 2
E         Left contains one more item: 2
E         Use -v to get the full diff

asdf/tests/test_fixture.py:10: AssertionError
```

Reply:
Claim red-herring argument! This is true for any fixture scoped greater than function. If a fixture is being misused, then expect the consequences.  Current fixture is creating resources that is appropriate for the test in use. 🐟